python/src/lib.rs

```diff
@@ -360,3 +368,48 @@ fn manifest_needs_migration(dataset: &Bound<'_, PyAny>) -> PyResult<bool> {
         &manifest, &indices,
     ))
 }
+
+#[pyclass(name = "FFILanceTableProvider", module = "lance", subclass)]
+#[derive(Clone)]
+struct FFILanceTableProvider {
+    dataset: Arc<::lance::Dataset>,
+    with_row_id: bool,
+    with_row_addr: bool,
+}
+
+#[pymethods]
+impl FFILanceTableProvider {
+    #[new]
+    #[pyo3(signature = (dataset, param_with_row_id = false, param_with_row_addr = false))]
+    fn new(
+        dataset: &Bound<'_, PyAny>,
+        param_with_row_id: bool,
+        param_with_row_addr: bool,
+    ) -> PyResult<Self> {
```

Comment:
I might suggest tweaking this slightly so that `param_with_row_id` and `param_with_row_addr` are keyword only arguments with a default of `False` and perhaps also rename them to `with_row_id` and `with_row_addr`.   However, we can do that in a follow-up.

Reply:
sure, done in commit https://github.com/lancedb/lance/pull/3649/commits/d39fc06f91cc05475186dbe730606507676418a4